File: Frontend/src/app/teachers/teacher.dto.ts
import {TeacherCourseDTO} from '../teacher-Course/teacher-Course.dto';

export interface TeacherDTO {
  id?: number;
  firstname: string;
  lastname: string;
  email: string;
  number: string;
  teacherCourses?: TeacherCourseDTO[];
}

File: Backend/src/main/java/projectir4/elearning/dto/TeacherDTO.java
package projectir4.elearning.dto;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class TeacherDTO {

    private Long id;
    private String username;
    private String email;
    private String number;
    private List<TeacherCourseDTO> teacherCourses;
}